test/fixtures/process-exit-code-cases.js

```diff
@@ -63,8 +63,14 @@ function getTestCases(isWorker = false) {
 
   function zeroExitWithUncaughtHandler() {
     process.on('exit', (code) => {
-      assert.strictEqual(process.exitCode, 0);
-      assert.strictEqual(code, 0);
+      if (code !== 0) {
+        console.log('wrong code! expected 0; got', code);
+        process.exit(99);
+      }
```

Comment:
```suggestion
      try {
        assert.strictEqual(process.exitCode, 0);
      } catch (err) {
        // We need to catch the error and exit in case there's an uncaughtException handler.
        console.error(error);
        process.exit(99);
      }
```